There is a bug in how jsonToStruct parses maps and slices.

There is a bug in how jsonToStruct parses maps and slices.
1. Does not support parsing map type
2. Parse the slice type, if the data type is inconsistent, panic error

```go
type Target struct {
	Followed map[string]bool `json:"followed"`
	Fids     []string        `json:"fids"`
}

func TestJsonToStruct(t *testing.T) {
	data := `
	{
		"followed": {
			"43015653": true,
			"43015666": false
		},
		"fids": ["43015653",43015666]
	}
	`

	var event Target
	err := JsonToStruct(data, &event)
	fmt.Printf("%+v, err:%s \n", event, err)
}
```

The specific error description is as follows

```
panic: reflect.Set: value of type float64 is not assignable to type string [recovered]
	panic: reflect.Set: value of type float64 is not assignable to type string
```

--- a/jsonUtil/Document.md
+++ b/jsonUtil/Document.md
@@ -42,3 +42,59 @@ func main() {
 ```
 
 在上面的例子中，我们首先将结构体指针 `p` 转换为 `reflect.Value` 类型的值 `v`，然后使用 `v.Elem()` 获取指针所指向的值，接着使用 `FieldByName()` 方法获取结构体字段的 `reflect.Value` 类型的值，最后使用 `SetString()` 和 `SetInt()` 方法修改结构体字段的值。
+
+## 处理Slice数据类型
+
+```go
+case reflect.Slice:
+    if subData, ok := value.([]interface{}); ok {
+        if err := parseSlice(fieldValue, subData); err != nil {
+            return err
+        }
+    } else {
+        return fmt.Errorf("unexpected value type for slice: %T", value)
+    }
+```
+
+将每个子元素都转换为结构体定义的数据类型再赋值给对应的字段。如果子元素是一个切片类型，需要进行递归处理。
+
+首先获取到切片的元素类型 subType，如果 subType 是一个指针类型，需要将其转换为实际的元素类型。然后创建一个与原始切片相同长度的新切片 subSliceValue，并循环遍历原始切片中的每个子元素。在每个子元素中，需要将其转换为结构体定义的数据类型，可以通过递归调用 parseStruct 或 parseSlice 来实现。最后，将新的子元素设置到新切片 subSliceValue 中，并将其赋值给对应的字段。注意，如果元素类型是一个结构体，需要使用 Set 方法设置到新切片中，如果是基本类型，则需要判定类型然后添加到值中。
+
+## 处理Map数据类型
+
+**第一种更加简洁的调用方法**
+
+```go
+case reflect.Map:
+    if subData, ok := value.(map[string]interface{}); ok {
+        subResult := reflect.New(fieldType.Type).Elem()
+        if err := parseMap(subResult, subData); err != nil {
+            return err
+        }
+        fieldValue.Set(subResult)
+    }
+```
+
+其中 subResult 使用 reflect.New 创建一个新的结构体对象，然后使用 Elem() 方法获取其指针所指向的值，最后传递给 parseMap 方法进行解析。这样可以避免对 reflect.MakeMap 的使用，更加简洁。
+
+
+**第二种实在原有的代码进行更改。**
+```go
+case reflect.Map:
+    if subData, ok := value.(map[string]interface{}); ok {
+        subResult := reflect.MakeMap(fieldValue.Type())
+        for subKey, subValue := range subData {
+            subElem := reflect.New(fieldType.Type.Elem())
+            err := parseValue(subElem.Elem(), subValue)
+            if err != nil {
+                return err
+            }
+            subResult.SetMapIndex(reflect.ValueOf(subKey), subElem.Elem())
+        }
+        fieldValue.Set(subResult)
+    }
+```
+
+其中，subData 是一个 map[string]interface{} 类型的变量，表示一个嵌套的 JSON 对象；subResult 是一个 reflect.Value 类型的变量，表示一个 map 类型的变量的反射值；subKey 和 subValue 分别表示 subData 中的键和值，可以通过 range 循环遍历 subData。
+
+在调用 parseValue 函数时，需要传入 subElem.Elem() 作为第一个参数，这是因为 subElem 表示一个指向 fieldType.Type.Elem() 类型变量的指针，而 parseValue 函数需要传入一个值类型的变量。所以需要通过 subElem.Elem() 获取到 fieldType.Type.Elem() 类型的变量。

--- a/jsonUtil/json_to_struct.go
+++ b/jsonUtil/json_to_struct.go
@@ -3,16 +3,26 @@ package jsonUtil
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 	"log"
 	"reflect"
 	"strings"
+)
+
+var (
+	// ErrPoint 不是指针类型
+	ErrPoint = errors.New("the argument to Result must be a non-nil pointer")
+	// ErrNotMap 不是Map类型
+	ErrNotMap = errors.New("cannot parse map, value is not a map")
+	// ErrNotSlice 不是Slice类型
+	ErrNotSlice = errors.New("cannot parse slice, value is not a slice")
 )
 
 // JsonToStruct Parses JSON into a specified structure pointer
 // 将JSON解析为指定的结构体指针
 func JsonToStruct(jsonData string, result any) error {
 	if reflect.ValueOf(result).Kind() != reflect.Pointer || reflect.ValueOf(result).IsNil() {
-		return errors.New("the argument to Result must be a non-nil pointer")
+		return ErrPoint
 	}
 
 	var data map[string]any
@@ -29,15 +39,8 @@ func JsonToStruct(jsonData string, result any) error {
 		fieldName := fieldType.Name
 		fieldValue := resultValue.FieldByName(fieldName)
 
-		// 从json的tag标签中取出定义字段
-		jsonTag := fieldType.Tag.Get("json")
-		if jsonTag == "" {
-			jsonTag = fieldName
-		} else {
-			if commaIndex := strings.Index(jsonTag, ","); commaIndex != -1 {
-				jsonTag = jsonTag[:commaIndex]
-			}
-		}
+		// 获取json的tag
+		jsonTag := getJsonTag(fieldType, fieldName)
 
 		value, ok := data[jsonTag]
 		if !ok {
@@ -80,33 +83,38 @@ func JsonToStruct(jsonData string, result any) error {
 				}
 				fieldValue.Set(subResult.Elem())
 			}
-		case reflect.Slice:
-			if subData, ok := value.([]any); ok {
-				subResult := reflect.MakeSlice(fieldValue.Type(), len(subData), len(subData))
-				for j := 0; j < len(subData); j++ {
-					subValue := subData[j]
-					subElem := subResult.Index(j)
-
-					if subElem.Kind() == reflect.Struct {
-						if subDataElem, ok := subValue.(map[string]any); ok {
-							subResultElem := reflect.New(subElem.Type())
-							err := JsonToStruct(convertToJSONString(subDataElem), subResultElem.Interface())
-							if err != nil {
-								return err
-							}
-							subElem.Set(subResultElem.Elem())
-						}
-					} else {
-						subElem.Set(reflect.ValueOf(subValue))
-					}
+		case reflect.Map:
+			if subData, ok := value.(map[string]any); ok {
+				subResult := reflect.New(fieldType.Type).Elem()
+				if err := parseMap(subResult, subData); err != nil {
+					return err
 				}
 				fieldValue.Set(subResult)
 			}
-			//default:
-			//	fieldValue.Set(reflect.ValueOf(value))
+		case reflect.Slice:
+			if subData, ok := value.([]any); ok {
+				if err := parseSlice(fieldValue, subData); err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("unexpected value type for slice: %T", value)
+			}
 		}
 	}
 	return nil
+}
+
+// 从json的tag标签中取出定义字段
+func getJsonTag(fieldType reflect.StructField, fieldName string) string {
+	jsonTag := fieldType.Tag.Get("json")
+	if jsonTag == "" {
+		jsonTag = fieldName
+	} else {
+		if commaIndex := strings.Index(jsonTag, ","); commaIndex != -1 {
+			jsonTag = jsonTag[:commaIndex]
+		}
+	}
+	return jsonTag
 }
 
 func convertToJSONString(data map[string]any) string {

--- a/jsonUtil/json_to_struct_map_test.go
+++ b/jsonUtil/json_to_struct_map_test.go
@@ -1,0 +1,71 @@
+package jsonUtil
+
+import (
+	"testing"
+)
+
+func TestJsonToStructMap1(t *testing.T) {
+	data := `
+	{
+		"uid": 43015653,
+		"foll": {
+			"43015653": true,
+			"43015666": false
+		},
+		"followed": {
+			"friendRed": {
+				"43015653": true,
+				"43015666": false
+			},
+			"friendWhite": {
+				"43015653": true,
+				"43015666": false
+			}
+		}
+	}
+	`
+	var target struct {
+		Uid      int                        `json:"uid"`
+		Foll     map[string]bool            `json:"foll"`
+		Followed map[string]map[string]bool `json:"followed"`
+	}
+
+	err := JsonToStruct(data, &target)
+	if err != nil {
+		t.Errorf("err %s", err)
+		return
+	}
+}
+
+func TestJsonToStructMap2(t *testing.T) {
+	data := `
+	{
+		"uid": 43015653,
+		"foll": {
+			"boy": {
+				"t1": "v1",
+				"t2": "v2"
+			},
+			"girl": {
+				"t1": "v1",
+				"t2": "v2"
+			}
+		}
+	}
+	`
+	type Val struct {
+		T1 string `json:"t1"`
+		T2 string `json:"t2"`
+	}
+
+	var target struct {
+		Uid  int            `json:"uid"`
+		Foll map[string]Val `json:"foll"`
+	}
+
+	err := JsonToStruct(data, &target)
+	if err != nil {
+		t.Errorf("err %s", err)
+		return
+	}
+}

--- a/jsonUtil/json_to_struct_slice_test.go
+++ b/jsonUtil/json_to_struct_slice_test.go
@@ -1,0 +1,24 @@
+package jsonUtil
+
+import (
+	"testing"
+)
+
+func TestJsonToStructSlice1(t *testing.T) {
+	data := `
+		{
+			"uid": 43015653,
+			"fids": ["43015653", 43015666]
+		}
+	`
+	var target struct {
+		Uid  int      `json:"uid"`
+		Fids []string `json:"fids"`
+	}
+
+	err := JsonToStruct(data, &target)
+	if err != nil {
+		t.Errorf("err %s", err)
+		return
+	}
+}

--- a/jsonUtil/json_to_struct_test.go
+++ b/jsonUtil/json_to_struct_test.go
@@ -3,6 +3,7 @@ package jsonUtil
 import (
 	"encoding/json"
 	"fmt"
+	"reflect"
 	"testing"
 )
 
@@ -311,6 +312,109 @@ func TestJsonToStructMoreNest(t *testing.T) {
 		if resultPerson.Emails[i] != expectedEmail {
 			t.Errorf("Emails[%d] mismatch: expected %s but got %s", i, expectedEmail, resultPerson.Emails[i])
 		}
+	}
+}
+
+// 合法验证：多层级嵌套
+func TestJsonToStructMoreNest2(t *testing.T) {
+	var jsonData = `
+	{
+		"uid": 666,
+		"use_id": ["hello", 5, 9],
+		"age": "20",
+		"equip": {
+			"keyMike": true,
+			"keyTom": false
+		},
+		"happy": {
+			"k1": [1, 2, 3],
+			"k2": [4, 5, 6]
+		},
+		"slices": [{
+			"nickname": "ABC",
+			"money": "20"
+		}, {
+			"nickname": "EFG",
+			"money": "22"
+		}],
+		"maps": {
+			"m1": {
+				"name": "alis",
+				"age": "20"
+			},
+			"m2": {
+				"name": "jom",
+				"age": "22"
+			}
+		}
+	}
+	`
+
+	type SliceVal struct {
+		Nickname string `json:"nickname"`
+		Money    int    `json:"money"`
+	}
+
+	type MapVal struct {
+		Name string `json:"name"`
+		Age  int    `json:"age"`
+	}
+
+	type Target struct {
+		Uid    int               `json:"uid"`
+		UseId  []string          `json:"use_id"`
+		Age    int               `json:"age"`
+		Equip  map[string]bool   `json:"equip"`
+		Happy  map[string][]int  `json:"happy"`
+		Slices []SliceVal        `json:"slices"`
+		Maps   map[string]MapVal `json:"maps"`
+	}
+
+	var res Target
+	err := JsonToStruct(jsonData, &res)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	//jsonRes, _ := json.Marshal(res)
+	//fmt.Printf("%+v \n", res)
+	//fmt.Println(string(jsonRes))
+
+	expected := Target{
+		Uid:   666,
+		UseId: []string{"hello", "5", "9"},
+		Age:   20,
+		Equip: map[string]bool{
+			"keyMike": true,
+			"keyTom":  false,
+		},
+		Happy: map[string][]int{
+			"k1": {1, 2, 3},
+			"k2": {4, 5, 6},
+		},
+		Slices: []SliceVal{
+			{
+				Nickname: "ABC",
+				Money:    20,
+			},
+			{
+				Nickname: "EFG",
+				Money:    22,
+			},
+		},
+		Maps: map[string]MapVal{
+			"m1": {
+				Name: "alis",
+				Age:  20,
+			},
+			"m2": {
+				Name: "jom",
+				Age:  22,
+			},
+		},
+	}
+
+	if !reflect.DeepEqual(res, expected) {
+		t.Errorf("Result not as expected.\nExpected: %+v\nActual: %+v\n", expected, res)
 	}
 }
 

--- a/jsonUtil/parse_map.go
+++ b/jsonUtil/parse_map.go
@@ -1,0 +1,122 @@
+package jsonUtil
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+func parseMap(value reflect.Value, data map[string]any) error {
+	if value.Kind() != reflect.Map {
+		return ErrNotMap
+	}
+
+	valueType := value.Type().Elem()
+	newMap := reflect.MakeMapWithSize(value.Type(), len(data))
+
+	for key, val := range data {
+		newKey := reflect.ValueOf(key)
+
+		if valueType.Kind() == reflect.Interface {
+			newMap.SetMapIndex(newKey, reflect.ValueOf(val))
+		} else if valueType.Kind() == reflect.Map {
+			newElem := reflect.New(valueType).Elem()
+			if err := parseMap(newElem, val.(map[string]any)); err != nil {
+				return err
+			}
+			newMap.SetMapIndex(newKey, newElem)
+		} else {
+			newVal := reflect.New(valueType).Elem()
+			if err := parseValue(newVal, val); err != nil {
+				return err
+			}
+			newMap.SetMapIndex(newKey, newVal)
+		}
+	}
+
+	value.Set(newMap)
+	return nil
+}
+
+func parseValue(fieldVal reflect.Value, v any) error {
+	switch fieldVal.Kind() {
+	case reflect.String:
+		str, ok := v.(string)
+		if !ok {
+			return errors.New("failed to parse string")
+		}
+		fieldVal.SetString(str)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		n, err := toInt64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetInt(n)
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		n, err := toUint64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetUint(uint64(n))
+	case reflect.Float32, reflect.Float64:
+		n, err := toFloat64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetFloat(n)
+	case reflect.Bool:
+		b, ok := v.(bool)
+		if !ok {
+			return errors.New("failed to parse bool")
+		}
+		fieldVal.SetBool(b)
+	case reflect.Struct:
+		if subData, ok := v.(map[string]any); ok {
+			if err := JsonToStruct(convertToJSONString(subData), fieldVal.Addr().Interface()); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("unexpected value type for struct: %T", v)
+		}
+	case reflect.Slice:
+		if arr, ok := v.([]any); ok {
+			elemType := fieldVal.Type().Elem()
+			sliceVal := reflect.MakeSlice(fieldVal.Type(), len(arr), len(arr))
+			for i, elem := range arr {
+				elemVal := reflect.New(elemType).Elem()
+				if err := parseValue(elemVal, elem); err != nil {
+					return err
+				}
+				sliceVal.Index(i).Set(elemVal)
+			}
+			fieldVal.Set(sliceVal)
+		} else {
+			return fmt.Errorf("unexpected value type for slice: %T", v)
+		}
+	case reflect.Map:
+		if mapData, ok := v.(map[string]any); ok {
+			mapType := fieldVal.Type()
+			if fieldVal.IsNil() {
+				fieldVal.Set(reflect.MakeMap(mapType))
+			}
+			elemType := mapType.Elem()
+			mapVal := fieldVal
+			for k, v := range mapData {
+				keyVal := reflect.New(mapType.Key()).Elem()
+				if err := parseValue(keyVal, k); err != nil {
+					return err
+				}
+				elemVal := reflect.New(elemType).Elem()
+				if err := parseValue(elemVal, v); err != nil {
+					return err
+				}
+				mapVal.SetMapIndex(keyVal, elemVal)
+			}
+		} else {
+			return fmt.Errorf("unexpected value type for map: %T", v)
+		}
+	default:
+		return fmt.Errorf("unsupported kind: %s", fieldVal.Kind())
+	}
+	return nil
+}

--- a/jsonUtil/parse_map_test.go
+++ b/jsonUtil/parse_map_test.go
@@ -1,0 +1,1 @@
+package jsonUtil

--- a/jsonUtil/parse_primitive_value.go
+++ b/jsonUtil/parse_primitive_value.go
@@ -1,0 +1,52 @@
+package jsonUtil
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+func parsePrimitiveValue(fieldVal reflect.Value, v any) error {
+	switch fieldVal.Kind() {
+	case reflect.String:
+		str, ok := v.(string)
+		if !ok {
+			return errors.New("failed to parse string")
+		}
+		fieldVal.SetString(str)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		n, err := toInt64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetInt(n)
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		n, err := toUint64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetUint(uint64(n))
+	case reflect.Float32, reflect.Float64:
+		n, err := toFloat64Reflect(v)
+		if err != nil {
+			return err
+		}
+		fieldVal.SetFloat(n)
+	case reflect.Bool:
+		b, ok := v.(bool)
+		if !ok {
+			return errors.New("failed to parse bool")
+		}
+		fieldVal.SetBool(b)
+	default:
+		return fmt.Errorf("unsupported kind: %s", fieldVal.Kind())
+	}
+	return nil
+}
+
+//case reflect.String, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+//reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+//reflect.Float32, reflect.Float64, reflect.Bool:
+//if err := parsePrimitiveValue(fieldVal, v); err != nil {
+//return err
+//}

--- a/jsonUtil/parse_primitive_value_test.go
+++ b/jsonUtil/parse_primitive_value_test.go
@@ -1,0 +1,85 @@
+package jsonUtil
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestParsePrimitiveValue(t *testing.T) {
+	// Test cases for reflect.String
+	strFieldVal := reflect.ValueOf(new(string)).Elem()
+	err := parsePrimitiveValue(strFieldVal, "hello")
+	if err != nil {
+		t.Errorf("Error parsing string value: %v", err)
+	}
+	if strFieldVal.String() != "hello" {
+		t.Errorf("Expected %q, got %q", "hello", strFieldVal.String())
+	}
+	err = parsePrimitiveValue(strFieldVal, 123)
+	if err == nil {
+		t.Error("Expected error for parsing int value as string")
+	}
+
+	// Test cases for reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64
+	intFieldVal := reflect.ValueOf(new(int64)).Elem()
+	err = parsePrimitiveValue(intFieldVal, "123")
+	if err != nil {
+		t.Errorf("Error parsing int value: %v", err)
+	}
+	if intFieldVal.Int() != 123 {
+		t.Errorf("Expected %d, got %d", 123, intFieldVal.Int())
+	}
+	err = parsePrimitiveValue(intFieldVal, "abc")
+	if err == nil {
+		t.Error("Expected error for parsing non-int value as int")
+	}
+
+	// Test cases for reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64
+	uintFieldVal := reflect.ValueOf(new(uint64)).Elem()
+	err = parsePrimitiveValue(uintFieldVal, "123")
+	if err != nil {
+		t.Errorf("Error parsing uint value: %v", err)
+	}
+	if uintFieldVal.Uint() != 123 {
+		t.Errorf("Expected %d, got %d", 123, uintFieldVal.Uint())
+	}
+	err = parsePrimitiveValue(uintFieldVal, 123)
+	if err != nil {
+		t.Errorf("Expected error  uint, err:%v", err)
+	}
+
+	// Test cases for reflect.Float32, reflect.Float64
+	floatFieldVal := reflect.ValueOf(new(float64)).Elem()
+	err = parsePrimitiveValue(floatFieldVal, "3.14")
+	if err != nil {
+		t.Errorf("Error parsing float value: %v", err)
+	}
+	if floatFieldVal.Float() != 3.14 {
+		t.Errorf("Expected %f, got %f", 3.14, floatFieldVal.Float())
+	}
+	err = parsePrimitiveValue(floatFieldVal, "abc")
+	if err == nil {
+		t.Error("Expected error for parsing non-float value as float")
+	}
+
+	// Test cases for reflect.Bool
+	boolFieldVal := reflect.ValueOf(new(bool)).Elem()
+	err = parsePrimitiveValue(boolFieldVal, true)
+	if err != nil {
+		t.Errorf("Error parsing bool value: %v", err)
+	}
+	if boolFieldVal.Bool() != true {
+		t.Errorf("Expected %t, got %t", true, boolFieldVal.Bool())
+	}
+	err = parsePrimitiveValue(boolFieldVal, "abc")
+	if err == nil {
+		t.Error("Expected error for parsing non-bool value as bool")
+	}
+
+	// Test cases for unsupported kind
+	unsupportedFieldVal := reflect.ValueOf(new([]string)).Elem()
+	err = parsePrimitiveValue(unsupportedFieldVal, "abc")
+	if err == nil {
+		t.Error("Expected error for unsupported kind")
+	}
+}

--- a/jsonUtil/parse_slice.go
+++ b/jsonUtil/parse_slice.go
@@ -1,0 +1,81 @@
+package jsonUtil
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"reflect"
+)
+
+func parseSlice(fieldValue reflect.Value, subData []any) error {
+	if fieldValue.Kind() != reflect.Slice {
+		return ErrNotSlice
+	}
+
+	// 获取切片元素类型
+	elemType := fieldValue.Type().Elem()
+
+	// 创建一个与目标字段相同类型的新切片
+	slice := reflect.MakeSlice(fieldValue.Type(), 0, len(subData))
+
+	// 迭代 subData 切片的元素
+	for _, item := range subData {
+		// 创建 slice 元素类型的新元素
+		elem := reflect.New(elemType).Elem()
+
+		// 如果元素类型是结构体，则递归调用 JsonToStruct
+		if elem.Kind() == reflect.Struct {
+			// 将项目转换为 JSON 字符串
+			jsonData, err := json.Marshal(item)
+			if err != nil {
+				return err
+			}
+
+			// 调用 JsonToStruct 将 JSON 字符串解析为 struct 元素
+			err = JsonToStruct(string(jsonData), elem.Addr().Interface())
+			if err != nil {
+				return err
+			}
+		} else {
+			// 否则，将项目转换为适当的类型并设置元素值
+			switch elem.Kind() {
+			case reflect.String:
+				elem.SetString(toStringReflect(item))
+			case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+				val, err := toInt64Reflect(item)
+				if err != nil {
+					log.Printf("toInt64Reflect err:%s \n", err)
+					return err
+				}
+				elem.SetInt(val)
+			case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+				val, err := toUint64Reflect(item)
+				if err != nil {
+					log.Printf("toUint64Reflect err:%s \n", err)
+					return err
+				}
+				elem.SetUint(val)
+			case reflect.Float32, reflect.Float64:
+				val, err := toFloat64Reflect(item)
+				if err != nil {
+					log.Printf("toFloat64Reflect err:%s \n", err)
+					return err
+				}
+				elem.SetFloat(val)
+			case reflect.Bool:
+				val := toBoolReflect(item)
+				elem.SetBool(val)
+			default:
+				return fmt.Errorf("unsupported slice element type: %s", elemType.String())
+			}
+		}
+
+		// 将元素附加到切片
+		slice = reflect.Append(slice, elem)
+	}
+
+	// 将切片值设置为目标字段
+	fieldValue.Set(slice)
+
+	return nil
+}

--- a/jsonUtil/parse_slice_test.go
+++ b/jsonUtil/parse_slice_test.go
@@ -1,0 +1,56 @@
+package jsonUtil
+
+import (
+	"reflect"
+	"testing"
+)
+
+type ExampleStruct struct {
+	Name string `json:"name"`
+	Age  int    `json:"age"`
+}
+
+func TestParseSlice(t *testing.T) {
+	var slice []any
+
+	// 添加元素到切片
+	slice = append(slice, "hello")
+	slice = append(slice, "world")
+
+	// 创建一个新的切片值
+	sliceValue := reflect.ValueOf(&[]string{}).Elem()
+
+	// 解析切片
+	err := parseSlice(sliceValue, slice)
+	if err != nil {
+		t.Errorf("expected no error, but got %s", err.Error())
+	}
+
+	// 检查解析的切片是否与预期相等
+	expectedSlice := []string{"hello", "world"}
+	actualSlice := sliceValue.Interface().([]string)
+	if !reflect.DeepEqual(expectedSlice, actualSlice) {
+		t.Errorf("expected %v, but got %v", expectedSlice, actualSlice)
+	}
+
+	// 测试解析结构体切片
+	var structSlice []any
+	structSlice = append(structSlice, ExampleStruct{"Tom", 30})
+	structSlice = append(structSlice, ExampleStruct{"Jerry", 25})
+
+	// 创建一个新的切片值
+	structSliceValue := reflect.ValueOf(&[]ExampleStruct{}).Elem()
+
+	// 解析切片
+	err = parseSlice(structSliceValue, structSlice)
+	if err != nil {
+		t.Errorf("expected no error, but got %s", err.Error())
+	}
+
+	// 检查解析的切片是否与预期相等
+	expectedStructSlice := []ExampleStruct{{"Tom", 30}, {"Jerry", 25}}
+	actualStructSlice := structSliceValue.Interface().([]ExampleStruct)
+	if !reflect.DeepEqual(expectedStructSlice, actualStructSlice) {
+		t.Errorf("expected %v, but got %v", expectedStructSlice, actualStructSlice)
+	}
+}

